### ISSUE TITLE
Make it easier to include "standard" types in the "google.protobuf" package

### ISSUE THREAD

**Issue by seh** [os72/protoc-jar-maven-plugin#5]
As of Protocol Buffers version 3, the compiler and runtime suite [ship with several "standard" types in the `google.protobuf` package](https://github.com/google/protobuf/tree/master/src/google/protobuf), such as `Timestamp` (in file [_google/protobuf/timestamp.proto_](https://github.com/google/protobuf/blob/master/src/google/protobuf/timestamp.proto)) and `Duration` (in file [_google/protobuf/duration.proto_](https://github.com/google/protobuf/blob/master/src/google/protobuf/duration.proto)). These files are not automatically available to the compiler, though.

The _protoc_ Maven plugin accepts any number of `includeDirectories` elements to add directories to the compiler's include path. However, the files in question are packaged into Jar files, such as [`com.google.protobuf:protobuf-java:3.0.0-alpha-3`](http://mvnrepository.com/artifact/com.google.protobuf/protobuf-java/3.0.0-alpha-3).

Is there some way that I can point the _protoc_ compiler at a file embedded in a Jar file? If not, would you consider unpacking these files from the Jar file (perhaps to a temporary directory, or to the "generated sources" directory tree) and making them available on the include path?

**Comment by seh** [comment]
I found a way to get the files into place, but it's not pretty:
```xml
<plugins>
  <build>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-dependency-plugin</artifactId>
      <executions>
        <execution>
          <id>copy-standard-protobuf-sources</id>
          <phase>generate-sources</phase>
          <goals>
            <goal>unpack</goal>
          </goals>
          <configuration>
            <artifactItems>
              <artifactItem>
                <groupId>com.google.protobuf</groupId>
                <artifactId>protobuf-java</artifactId>
                <type>jar</type>
                <overWrite>true</overWrite>
                <outputDirectory>${project.build.directory}/generated-sources/protobuf</outputDirectory>
                <includes>**/*.proto</includes>
              </artifactItem>
            </artifactItems>
          </configuration>
        </execution>
        <execution>
          <id>analyze</id>
          <goals>
            <goal>analyze-only</goal>
          </goals>
          <configuration>
            <failOnWarning>true</failOnWarning>
          </configuration>
        </execution>
      </executions>
    </plugin>
  </plugins>
</build>
```
Then, in the configuration for the _protoc_ plugin:
``` xml
<configuration>
  <includeDirectories>
    <include>${project.build.directory}/generated-sources/protobuf</include>
  </includeDirectories>
</configuration>
```
Hence, it's not impossible, but again, it could be easier.

**Comment by os72** [comment]
Sorry, I've missed/ignored this..

How do you imagine specifying it?
```xml
<addProtos>
  <artifactId>protobuf-java</artifactId>
  <artifactId>some-other-protos</artifactId>
  ...
</addProtos>
```
?

**Comment by seh** [comment]
(Coming back to this many months later...)

We'd need full Maven coordinates there, not just the artifact ID, and the possibility to apply a file filter to each. In the one artifact I had to expand, I wanted _all_ the "proto" files, but I can imagine others where you'd only want a subset of whatever is in the artifact.

**Comment by os72** [comment]
Decided to bundle the google.protobuf types directly with the plugin. (Google bundles them with the binary protoc distro) Set the new parameter `includeStdTypes` to `true` and the plugin will extract them and populate the protoc import path accordingly

**Comment by seh** [comment]
Fantastic. Thank you. I can now remove [my ugly workaround](https://github.com/os72/protoc-jar-maven-plugin/issues/5#issuecomment-134376616).
I just realized that I had filed this issue just over a year ago, also while "on vacation" in the same place.

**Comment by os72** [comment]
Released 3.0.0.1
